securesystemslib/signer.py

```diff
@@ -137,8 +148,154 @@ def to_dict(self) -> Dict:
+        keyid: str,
+        keytype: str,
+        scheme: str,
+        keyval: Dict[str, Any],
+        unrecognized_fields: Optional[Dict[str, Any]] = None,
+    ):
+        if not all(
+            isinstance(at, str) for at in [keyid, keytype, scheme]
+        ) or not isinstance(keyval, dict):
+            raise TypeError("Unexpected Key attributes types!")
+        self.keyid = keyid
+        self.keytype = keytype
+        self.scheme = scheme
+        self.keyval = keyval
+        if unrecognized_fields is None:
+            unrecognized_fields = {}
+
+        self.unrecognized_fields = unrecognized_fields
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Key):
+            return False
+
+        return (
+            self.keyid == other.keyid
+            and self.keytype == other.keytype
+            and self.scheme == other.scheme
+            and self.keyval == other.keyval
+            and self.unrecognized_fields == other.unrecognized_fields
+        )
+
+    @classmethod
+    def from_dict(cls, keyid: str, key_dict: Dict[str, Any]) -> "Key":
+        """Creates ``Key`` object from TUF serialization dict.
+
+        Raises:
+            KeyError, TypeError: Invalid arguments.
+        """
+        keytype = key_dict.pop("keytype")
+        scheme = key_dict.pop("scheme")
+        keyval = key_dict.pop("keyval")
+        # All fields left in the key_dict are unrecognized.
+        return cls(keyid, keytype, scheme, keyval, key_dict)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Returns a dict for TUF serialization."""
+        return {
+            "keytype": self.keytype,
+            "scheme": self.scheme,
+            "keyval": self.keyval,
+            **self.unrecognized_fields,
+        }
+
+    def to_securesystemslib_key(self) -> Dict[str, Any]:
+        """Returns a classic Securesystemslib keydict"""
+        return {
+            "keyid": self.keyid,
+            "keytype": self.keytype,
+            "scheme": self.scheme,
+            "keyval": self.keyval,
+        }
+
+    @classmethod
+    def from_securesystemslib_key(cls, key_dict: Dict[str, Any]) -> "Key":
+        """Creates a ``Key`` object from a classic securesystemlib keydict.
+
+        Args:
+            key_dict: Key in securesystemlib dict representation.
+
+        Raises:
+            ValueError: ``key_dict`` value is not in securesystemslib format.
+        """
+        try:
+            key_meta = sslib_keys.format_keyval_to_metadata(
+                key_dict["keytype"],
+                key_dict["scheme"],
+                key_dict["keyval"],
+            )
+        except FormatError as e:
+            raise ValueError("keydict not in securesystemslib format") from e
+
+        return cls(
+            key_dict["keyid"],
+            key_meta["keytype"],
+            key_meta["scheme"],
+            key_meta["keyval"],
+        )
+
+    def is_verified(self, signature: Signature, data: bytes) -> bool:
```

Comment:
This requires a bit of thinking. What I did here is:
* leave `verify_signature()` in python-tuf (as it's metadata related)
* add new `is_verified()` to wrap `keys.verify_signature()`

The former raises on verification failure, latter returns False.

The other alternative would be to implement a `verify_signature()` here as well:a method that would also raise on failure, but would take bytes as input. This would require the overload in python-tuf to do some tricks: either implement the "overload dispatch" but that is not too onerous I think.

I think I like the  `verify_signature()` overload idea more than the PR here

Reply:
With "overload dispatch" I mean a setup like this for verify_signature:
```python
class Base:
    def func(self, i: int) -> None:
        # implement base func here

class Derived(Base):
    @overload
    def func(self, arg: int) -> None:
        ...

    @overload
    def func(self, arg: str, extra_arg: Optional[str] = None) -> None:
        ...

    def func(self, arg: Union[int, str], extra_arg: Optional[str] = None) -> None:
        if isinstance(arg, int):
            return super().func(arg)

        # implement derived func here


```

Reply:
> The other alternative would be to implement a verify_signature() here as well:a method that would also raise on failure like the namesake in python-tuf, but would take bytes as input. This would require the overload in python-tuf to do some tricks: likely implement the "overload dispatch" (or just silence mypy) but that seems reasonable..

Can you explain why this requires overloading? In your example, what is Base, Derived, func?

Reply:
Btw. there's a related issue about the return type of `Key.{is_verified aka. verify aka. verify_signature}` in https://github.com/in-toto/securesystemslib/issues/8